File: Frontend/composeApp/src/commonMain/kotlin/com/teamnotfound/airise/data/serializable/UserData.kt
package com.teamnotfound.airise.data.serializable

import androidx.compose.runtime.MutableState
import androidx.compose.runtime.mutableStateOf
import androidx.compose.ui.graphics.ImageBitmap
import kotlinx.serialization.Serializable

@Serializable
class UserData {
    var firstName: MutableState<String> = mutableStateOf("")
    var lastName: MutableState<String> = mutableStateOf("")
    var middleName: MutableState<String> = mutableStateOf("")
    var workoutGoal: MutableState<String> = mutableStateOf("")
    var fitnessLevel: MutableState<String> = mutableStateOf("")
    var workoutLength: MutableState<String> = mutableStateOf("")
    var equipmentAccess: MutableState<String> = mutableStateOf("")
    var workoutDays: MutableState<String> = mutableStateOf("")
    var workoutTime: MutableState<String> = mutableStateOf("")
    var dietaryGoal: MutableState<String> = mutableStateOf("")
    var workoutRestrictions: MutableState<String> = mutableStateOf("")
    var heightMetric: MutableState<Boolean> = mutableStateOf(false)
    var heightValue: MutableState<Int> = mutableStateOf(0)
    var weightMetric: MutableState<Boolean> = mutableStateOf(false)
    var weightValue: MutableState<Int> = mutableStateOf(0)
    var dobDay: MutableState<Int> = mutableStateOf(0)
    var dobMonth: MutableState<Int> = mutableStateOf(0)
    var dobYear: MutableState<Int> = mutableStateOf(0)
    var activityLevel: MutableState<String> = mutableStateOf("")

    // Convert to a serializable data class for MongoDB storage
    fun toData(): UserOnboardingData = UserOnboardingData(
        firstName = firstName.value, 
        lastName = lastName.value, 
        middleName = middleName.value,
        workoutGoal = workoutGoal.value,
        fitnessLevel = fitnessLevel.value,
        workoutLength = workoutLength.value,
        workoutEquipment = equipmentAccess.value,
        workoutDays = workoutDays.value,
        workoutTime = workoutTime.value,
        dietaryGoal = dietaryGoal.value,
        workoutRestrictions = workoutRestrictions.value,
        heightMetric = heightMetric.value,
        heightValue = heightValue.value,
        weightMetric = weightMetric.value,
        weightValue = weightValue.value,
        dobDay = dobDay.value,
        dobMonth = dobMonth.value,
        dobYear = dobYear.value,
        activityLevel = activityLevel.value
    )
}

@Serializable
data class UserOnboardingData(
    val firstName: String, 
    val lastName: String, 
    val middleName: String,
    val workoutGoal: String,
    val fitnessLevel: String,
    val workoutLength: String,
    val workoutEquipment: String,
    val workoutDays: String,
    val workoutTime: String,
    val dietaryGoal: String,
    val workoutRestrictions: String,
    val heightMetric: Boolean,
    val heightValue: Int,
    val weightMetric: Boolean,
    val weightValue: Int,
    val dobDay: Int,
    val dobMonth: Int,
    val dobYear: Int,
    val activityLevel: String
)

File: Backend/Model/User/UserData.cs
using System.Text.Json.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace AiRise.Models.User
{
    public class UserData
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("firebaseUid")]
        [JsonPropertyName("firebaseUid")]
        public string FirebaseUid { get; set; } = null!;

        [BsonElement("firstName")]
        [JsonPropertyName("firstName")]
        public string FirstName { get; set; } = string.Empty;

        [BsonElement("lastName")]
        [JsonPropertyName("lastName")]
        public string LastName { get; set; } = string.Empty;

        [BsonElement("middleName")]
        [JsonPropertyName("middleName")]
        public string MiddleName { get; set; } = string.Empty;
        
        [BsonElement("fullName")]
        [JsonPropertyName("fullName")]
        public string FullName { get; set; } = string.Empty;

        [BsonElement("workoutGoal")]
        [JsonPropertyName("workoutGoal")]
        public string WorkoutGoal { get; set; } = string.Empty;

        [BsonElement("fitnessLevel")]
        [JsonPropertyName("fitnessLevel")]
        public string FitnessLevel { get; set; } = string.Empty;

        [BsonElement("workoutLength")]
        [JsonPropertyName("workoutLength")]
        public int WorkoutLength { get; set; }

        [BsonElement("workoutEquipment")]
        [JsonPropertyName("workoutEquipment")]
        public string WorkoutEquipment { get; set; } = string.Empty;

        // array of day names
        [BsonElement("workoutDays")]
        [JsonPropertyName("workoutDays")]
        public List<string> WorkoutDays { get; set; } = new List<string>();

        [BsonElement("workoutTime")]
        [JsonPropertyName("workoutTime")]
        public string WorkoutTime { get; set; } = string.Empty;

        [BsonElement("dietaryGoal")]
        [JsonPropertyName("dietaryGoal")]
        public string DietaryGoal { get; set; } = string.Empty;

        [BsonElement("workoutRestrictions")]
        [JsonPropertyName("workoutRestrictions")]
        public string WorkoutRestrictions { get; set; } = string.Empty;

        [BsonElement("heightMetric")]
        [JsonPropertyName("heightMetric")]
        public bool HeightMetric { get; set; }

        [BsonElement("heightValue")]
        [JsonPropertyName("heightValue")]
        public int HeightValue { get; set; }

        [BsonElement("weightMetric")]
        [JsonPropertyName("weightMetric")]
        public bool WeightMetric { get; set; }

        [BsonElement("weightValue")]
        [JsonPropertyName("weightValue")]
        public int WeightValue { get; set; }

        [BsonElement("dobDay")]
        [JsonPropertyName("dobDay")]
        public int DobDay { get; set; }

        [BsonElement("dobMonth")]
        [JsonPropertyName("dobMonth")]
        public int DobMonth { get; set; }

        [BsonElement("dobYear")]
        [JsonPropertyName("dobYear")]
        public int DobYear { get; set; }

        [BsonElement("activityLevel")]
        [JsonPropertyName("activityLevel")]
        public string ActivityLevel { get; set; } = string.Empty;
    }
}